sdk/python/lib/pulumi/runtime/rpc.py

```diff
@@ -582,8 +583,41 @@ def translate_output_properties(output: Any,
     :param Optional[type] typ: The output's target type.
     :param bool transform_using_type_metadata: Set to True to use the metadata from `typ` to do name translation instead
                                                of using `output_transformer`.
+
+    :param Optional[Any]: Used internally to track recursive descent and enhance error messages.
     """
 
+
+    def sub_path(prop_name: str) -> Any:
```

Comment:
Should we use a stronger type than `Any` and dict? How about a [`NamedTuple`](https://docs.python.org/3/library/typing.html#typing.NamedTuple)? Or is it because we don't want to "expose" a new type?

Reply:
You know, I think this is my 'FP hacker' bad habits showing. I wanted an immutable linked list (to avoid quadratic `((([..] + [..]) + [..]) + [..]) + [..]` in case we ever have deep paths here, but Python does not have that in sdlib. I can clean this out a little and take it to private top-level _members.